nextest-runner/src/list/test_list.rs

```diff
@@ -676,6 +683,9 @@ impl<'g> RustTestArtifact<'g> {
             &self.package,
             dylib_path,
             &self.non_test_binaries,
+            // We don't need to set the user's custom environment variables defined in `config.toml`
+            // just to get the list of tests.
+            &Vec::new(),
```

Comment:
This isn't quite correct -- custom test harnesses might depend on these variables.